test/python/transpiler/test_target.py

```diff
@@ -1166,6 +1166,24 @@ def test_instruction_supported_no_args(self):
     def test_instruction_supported_no_operation(self):
         self.assertFalse(self.ibm_target.instruction_supported(qargs=(0,), parameters=[math.pi]))
 
+    def test_target_serialization_preserve_variadic(self):
+        """Checks that variadics are still seen as variadic after serialization"""
+        from pickle import loads, dumps
+
+        target = Target("test", 1)
+        # Add variadic example gate with no properties.
+        target.add_instruction(XGate, None, "x_var")
```

Comment:
It might be better to add a gate that doesn't have a fixed number of qubits, you can use `UnitaryGate` as a since if can have a variable number of qubits based on the size of the unitary matrix.